src/python.rs

```diff
@@ -1,62 +1,131 @@
-use crate::error;
+use crate::stacktrace::ProcessStacktrace;
+use crate::util::Exception;
+
+use crate::execution_class::ExecutionClass;
+use anyhow::{bail, Result};
 use gdb_command::stacktrace::*;
 use regex::Regex;
 
-/// Extract stack trace object from python traceback string
-///
-/// # Arguments
-///
-/// * `entries` - traceback as vector
-///
-/// # Return value
-///
-/// Traceback as a `Stacktrace` struct
-pub fn stacktrace_from_python(entries: &[String]) -> error::Result<Stacktrace> {
-    let mut stacktrace = Stacktrace::new();
-
-    for entry in entries.iter() {
-        let mut stentry = StacktraceEntry::default();
-
-        if entry.starts_with('[') {
-            let re = Regex::new(r#"\[Previous line repeated (\d+) more times\]"#).unwrap();
-            if let Some(rep) = re.captures(entry) {
-                let rep = rep.get(1).unwrap().as_str().parse::<u64>();
-                if rep.is_err() {
-                    return Err(error::Error::Casr(format!("Couldn't parse num: {entry}")));
-                }
-                let rep = rep.unwrap();
-                let last = stacktrace.last().unwrap().clone();
-                for _ in 0..rep {
-                    stentry = last.clone();
-                    stacktrace.push(stentry);
+pub struct PythonAnalysis;
```

Comment:
PythonStacktrace

Reply:
2 traits implemented for PythonAnalysis: Exception and ProcessStacktrace

Reply:
Use separate Structs PythonStacktrace and PythonException